superset/models/helpers.py

```diff
@@ -97,10 +97,13 @@ def formatter(c):
 
     @classmethod
     def import_from_dict(cls, session, dict_rep, parent=None,
-                         recursive=True, sync=[]):
+                         recursive=True, sync=[], respect_id=True):
```

Comment:
I am not sure of the changes in this entire file: They are working around test failures but I think they are hack and don't help. I will revert them once we have agreed about what to do about the failing tests